src/python/WMCore/MicroService/Unified/MSTransferor.py

```diff
@@ -1,102 +1,83 @@
+class MSTransferor(MSCore):
+    """
+    MSTransferor class provide whole logic behind
+    the transferor module.
+    """
+    def __init__(self, msConfig, logger=None):
         """
         Runs the basic setup and initialization for the MS Transferor module
         :param microConfig: microservice configuration
         """
-        self.msConfig = microConfig
-        self.status = status
-        self.uConfig = {}
-        self.logger = getMSLogger(microConfig['verbose'], logger=logger)
-        self.reqmgr2 = ReqMgr(microConfig['reqmgrUrl'], logger=self.logger)
-        self.reqmgrAux = ReqMgrAux(microConfig['reqmgrUrl'], httpDict={'cacheduration': 60}, logger=self.logger)
-        # eventually will change it to Rucio
-        self.phedex = PhEDEx(httpDict={'cacheduration': 10 * 60},
-                             dbsUrl=microConfig['dbsUrl'], logger=self.logger)
-
-    def prep(self):
-        """
-        Runs any preparation tasks before executing the actual algorithm.
-        For now:
-         * fetches the unified configuration
-        :return: False if it fail to run this action, otherwise True
-        """
-        self.uConfig = self.reqmgrAux.getUnifiedConfig(docName="config")
-        return bool(self.uConfig)
+        super(MSTransferor, self).__init__(msConfig, logger)
+        self.reqInfo = RequestInfo(msConfig, logger)
 
-    def execute(self):
+    def execute(self, reqStatus):
         """
         Executes the whole transferor logic
-        :param reqStatus: request status to that matters for this module
+        :param reqStatus: request statue to process
         :return:
         """
-        if not self.prep():
-            self.logger.warning("Failed to fetch the latest unified config. Skipping this cycle")
-            return
-        self.uConfig = self.uConfig[0]
-
         requestRecords = []
         try:
-            # get requests from ReqMgr2 data-service for given status
-            requests = self.reqmgr2.getRequestByStatus([self.status], detail=True)
-            if requests:
-                requests = requests[0]
-            self.logger.info("### transferor found %s requests in '%s' state", len(requests), self.status)
-            if requests:
-                for _, wfData in requests.iteritems():
+            # get requests from ReqMgr2 data-service for given statue
+            requestSpecs = self.reqmgr2.getRequestByStatus(
+                [reqStatus], detail=True)
+            if requestSpecs:
+                for _, wfData in requestSpecs[0].items():
                     requestRecords.append(self.requestRecord(wfData))
-        except Exception as err:
+            self.logger.debug(
+                '### transferor found %s requests in %s state',
+                len(requestRecords), reqStatus)
+            # get complete requests information
+            # based on Unified Transferor logic
+            requestRecords = self.reqInfo(requestRecords)
```

Comment:
You have just reverted the slicing changes I made. 
This code can potentially blow up memory usage on the node, since it collects tons of information for each request and only in the very end it would perform the data transfers. Please put the slicing logic back here.

Reply:
Alan, I was trying to merge broken branch and clean-up the conflicts. Please explicitly point out which slicing you're talking about and we can put it back. Better, I suggest to make another PR once this one will be merged since we first need to get proper codebase without conflicts.